js/kucoinfutures.js

```diff
@@ -1059,11 +1064,38 @@ module.exports = class kucoinfutures extends kucoin {
             'leverage': 1,
         };
         const stopPrice = this.safeValue2 (params, 'triggerPrice', 'stopPrice');
+        const stopLossPrice = this.safeValue (params, 'stopLossPrice');
+        const takeProfitPrice = this.safeValue (params, 'takeProfitPrice');
+        const isStopLoss = stopLossPrice !== undefined;
+        const isTakeProfit = takeProfitPrice !== undefined;
+        const stop = this.safeString (params, 'stop');
+        const stopPriceType = this.safeString (params, 'stopPriceType', 'MP');
         if (stopPrice) {
-            request['stop'] = (side === 'buy') ? 'up' : 'down';
-            const stopPriceType = this.safeString (params, 'stopPriceType', 'TP');
-            request['stopPriceType'] = stopPriceType;
+            if (stop !== undefined) {
```

Comment:
again we don't need to do this `request` will be extended with `params`, so you can replace this `if {} else{}` with 
```Javascript
if (stop === undefined) { 
      request['stop'] = (side === 'buy') ? 'down' : 'up';
}
```

Reply:
we actually don't need any check, because it will be overwritten